geode-core/src/main/java/org/apache/geode/internal/datasource/GemFireBasicDataSource.java

```diff
@@ -36,7 +36,7 @@
   private static final long serialVersionUID = -4010116024816908360L;
 
   /** Creates a new instance of BaseDataSource */
-  protected transient Driver driverObject = null;
+  protected transient volatile Driver driverObject = null;
```

Comment:
Did you mean to make this `volatile`? It seems unrelated.

Reply:
@jdeppe-pivotal 
The double checking is done in getConnection, Updating to use local variables too